.github/workflows/ci-build.yaml

```diff
@@ -1,7 +1,13 @@
 name: CI build action
 
 on:
-  pull_request:
+  # When a PR is opened from a fork repo, we want to be able to
+  # get permission to post comments on the PR. The workflow
+  # will be defined by the workflow definition *at HEAD*, not
+  # from the PR. When iterating on this workflow, you may want to
+  # change this to `pull_request` and create your PR from a
+  # branch on this repo, instead of from a fork.
+  pull_request_target:
```

Comment:
`tl;dr` Before we do this, let's narrow the scope of where we need write permissions.

Right now, the vast majority of workflows in this file have:
```yaml
permissions:
  contents: read
```

Which is great. However, the `build` workflow, which runs untrusted code asks for:
```yaml
permissions:
  contents: read
  pull-requests: write # so we have permission to comment on pull requests
```

Let's split up `build` into a phase that only needs `contents: read`:
1. Checks out the PR
2. Builds the APK(s)
3. Uploads the APK(s) to the PR

Then have a follow-on workflow that elevates to `pull-requests: write` and:
1. Downloads the APKs
2. Diffs the APKs against the latest build
3. Posts the diff to the PR

The key feature of this second workflow is that it does not execute any code from the checkout, which is not the case with this change as-is.